--- m17n-contrib-1.1.8/im/as-inscript.mim	2008-01-29 10:08:08.000000000 +0530
+++ m17n-contrib-1.1.8/im/as-inscript-new.mim	2008-04-02 13:54:17.000000000 +0530
@@ -41,8 +41,8 @@
 6. The alphabet 'আ' can be typed in two ways: one is to type 'অ' and then 'া' ,i.e. 'D' and then 'e'; other is by typing 'E' alone.
 7. The alphabet 'ঃ' is found in '_' key, i.e. 'Shift' and then '-' key in English keyboard.
 8. The alphabet 'ঋ' is located in '+', i.e. 'Shift' and then '=' key in English keyboard.
-9. The "Rakar" matra is typed as 'd' and then 'j'.
-10. The "ref" is typed as 'j' and then 'd' .
+9. The \"Rakar\" matra is typed as 'd' and then 'j'.
+10. The \"ref\" is typed as 'j' and then 'd' .
 11. Special characters 'জ্ঞ', 'ত্ৰ', 'ক্ষ' and 'শ্ৰ' are included respectively at '%', '^', '&' and '*'.
 12. The special combinations for 'ref' and 'rakar' are incorporated respectively at '$' and '#' .
 13. The character '৺' can be obtained by pressing 'Z', i.e. 'Shift' and 'z'.
